packages/material-ui/src/MenuItem/MenuItem.d.ts

```diff
@@ -1,11 +1,12 @@
 import * as React from 'react';
 import { StandardProps } from '..';
+import { ExtendButtonBase } from '../ButtonBase';
 import { ListItemProps, ListItemTypeMap } from '../ListItem';
-import { SimplifiedPropsOf, OverridableComponent } from '../OverridableComponent';
+import { SimplifiedPropsOf } from '../OverridableComponent';
 
 export type MenuItemClassKey = 'root' | 'gutters' | 'selected';
 
-declare const MenuItem: OverridableComponent<ListItemTypeMap<{ role?: string }, 'li'>>;
+declare const MenuItem: ExtendButtonBase<ListItemTypeMap<{ role?: string }, 'li'>>;
```

Comment:
Could you start by adding a typescript version of the react-select demo i.e. creating `IntegrationReactSelect.tsx` in `docs/src/pages/demos/autocomplete`. While this may fix the issue it doesn't look quite right.

Reply:
ExtendButtonBase was defined like this
```
export type ExtendButtonBase<M extends OverridableTypeMap> = ((
  props: { href: string } & OverrideProps<ExtendButtonBaseTypeMap<M>, 'a'>,
) => JSX.Element) &
  OverridableComponent<ExtendButtonBaseTypeMap<M>>;
```

I understand this way it `OverridableComponent` and some `Button` props.
That's exactly what we want here.

In this PR (https://github.com/mui-org/material-ui/commit/54f9a33b279fbd999a2e87e24b1ec6f7420242e2#diff-a242e1d972e33ce7b57d1b3296de00aa), all these generics types was used this way
```
ToggleButton: ExtendButtonBase<{ ....}>
```
For ButtonLike component, and
```
Avatar: OverridableComponent<{
```
For others.

Since MenuItem is a ButtonLike, I think it does look quite right.

Reply:
It isn't though. It has `ListItem` as its root component.

Reply:
Yes, but with `button` props set to `true`.
And then in ListItem implementation, 
```
if (button) {
  componentProps.component = componentProp || 'div';
  componentProps.focusVisibleClassName = clsx(classes.focusVisible, focusVisibleClassName);
  Component = ButtonBase;
}

....

return (
  <Component ref={ref} {...componentProps}>
    {children}
  </Component>
);
``` 

I'm not sure to understand your problem.

I made a new commit, is it better for you ?